client/public/index.html

```diff
@@ -27,17 +27,32 @@
       color="#5bbad5"
     />
     <meta name="theme-color" content="#ffffff" />
-    <!--
-      manifest.json provides metadata used when your web app is installed on a
-      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
-    -->
+
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
     <title>MDN Web Docs</title>
     <meta
       name="description"
       content="The MDN Web Docs site provides information about Open Web technologies including HTML, CSS, and APIs for both Web sites and progressive web apps."
     />
     <link rel="canonical" href="https://developer.mozilla.org" />
+    <style media="print">
+      .breadcrumbs-locale-container,
+      .document-toc-container,
+      .on-github,
+      nav.sidebar,
+      .page-header-main,
+      .page-footer,
+      ul.prev-next {
+        display: none !important;
```

Comment:
Do we need `!important`? I don't know if React injects some things when the file size is below a certain threshold into the page and so, perhaps we do 😞 

I know this is so little CSS but, could we make this an external stylesheet we import into `App.tsx` as the last piece of CSS? Perhaps `print-document.css` The thing I do not know is how to tell React that it is meant only for `media="print"` and there is a benefit to having that attribute set.

I am not _that_ concerned about having it in an external stylesheet although, it can grow over time. Perhaps we can add a comment here so when one searches the codebase for a print stylesheet this will come up? Then again, if you search for `media="print"` this will pop-up.

Ugh, I am torn. What do you think?

Reply:
The `!important` is there because the order of CSS loading is not guaranteed. First of all, it's nice to be able to not have to worry about how the rendering puts in the `<link>` tags. Secondly, hopefully we'll soon have a solution that injects the critical CSS as a `<style>` tag and then really lazy-load the external full `.css` file(s) after the DOM has loaded (the first time). That way, with the `!important` we never have to worry about the screen CSS and how/when it's injected into the DOM.

Reply:
Another option I was exploring was to use something like:
```
@media print {
    .toc { display:none }
}
```
in one of the `.scss` files. For example, in `client/src/document/index.tsx` this:
```diff
import "./index.scss";
+import "./print.scss";
```

But that will "bloat" the .css bundle. It's not too horrible but it's render-blocking after all so best to avoid if possible. 

Another option would be to use:
```html
<link rel="stylesheet" media="print" href="/static/css/print.7125b4e0a.css">
```
but the way it works in browsers, although lowest priority, it *always* downloads this. HTTP2 is fast but it's yet another thing that always has to be downloaded. 
Besides, I don't even know how to make that file! `react-scripts` will automatically turn all `import "foo.scss"` files into a bundle, through ,node-sass`, and save to `build/static/css/`. But you can't tell it to produce one separately. The only time you get multiple bundles is through `webpack` code-splitting but that won't help us here. 

Thankfully, any `<style>` tag you type into the `public/index.html` will automatically get rewritten by `react-scripts` to at least be whitespace optimized. 

Having it inline in the HTML bloats the HTML a bit which is also "render-blocking", obviously, since the rendering can't really start until the HTML file has been downloaded. 
But it's progressive render-blocking in that the HTML can actually start being worked on even before it's fully downloaded. And since the browser can quickly spot the `media="print"` it immediately knows it can ignore it for now. 

But the use case of being able to use SCSS is pretty good. Then you'd technically be able to express yourself better. For example, you'd be able to use variables, from another file, for things like `@media print { some-selector{ .$base-padding }}`. So perhaps it's better to have it in SCSS for that reason. But is that realistic?? Is there any desire or need to be able to author the print CSS in SCSS files?